src/main/java/flaxbeard/immersivepetroleum/common/blocks/metal/TileEntityPumpjack.java

```diff
@@ -146,6 +147,31 @@ public void update()
 		update(true);
 	}
 
+	// Checks whether or not this pump has a pipe to bedrock.
+	private boolean hasPipe() {
+		if (!IPConfig.Extraction.require_pipes) {
+			return true;
+		}
+		
+		World world = this.getWorld();
+		BlockPos pos = this.getOrigin();
+		int depthCount = 0;
+		int depthBounds = 256;
+		int pipeMeta = BlockTypes_MetalDevice1.FLUID_PIPE.getMeta();
+		while (depthCount < depthBounds) {
+			pos = pos.add(0, -1, 0);
+			IBlockState state = world.getBlockState(pos);
+			Block block = state.getBlock();
+			if (state == Blocks.BEDROCK.getDefaultState()) {
+				return true;
+			} else if (!(block instanceof BlockMetalDevice1 && block.getMetaFromState(state) == pipeMeta)) {
```

Comment:
I only want to return false if the block is not a fluid pipe, and return nothing otherwise. I could use this state check instead of the current meta check and hopefully that has better clarity.